# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/ScriptLoader.h
# Commit: 7770ec4717fd
# Full Hash: 7770ec4717fdce23c2e399fcbbd93a42a69c7721
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-12-21 21:21:23
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Disallow ImportScripts from module workers; r=jonco
#   
#   ImportScripts should be disallowed for module works, which are initialized in the following way:
#   `new Worker("url", { module:true})`. We set the WorkerType for workers accordingly, and can use that
#   to detect if import scripts are being incorrectly used.
# ==============================================================================

diff -r 68b476066248 -r 7770ec4717fd dom/workers/ScriptLoader.h
--- a/dom/workers/ScriptLoader.h	Tue Dec 20 20:56:12 2022 +0000
+++ b/dom/workers/ScriptLoader.h	Tue Dec 20 20:56:12 2022 +0000
@@ -185,7 +185,7 @@
                      nsISerialEventTarget* aSyncLoopTarget,
                      WorkerScriptType aWorkerScriptType, ErrorResult& aRv);
 
-  void CreateScriptRequests(const nsTArray<nsString>& aScriptURLs,
+  bool CreateScriptRequests(const nsTArray<nsString>& aScriptURLs,
                             const mozilla::Encoding* aDocumentEncoding,
                             bool aIsMainScript);
 
